### PR TITLE
Transform only svg files

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/babel-plugin-react-svg/src/index.js [hunk 1]
```diff
@@ -146,25 +146,30 @@ export default function(babel: BabelCore) {
...
-    visitor: Object.assign(
-      {},
-      programVisitor,
-      svgExpressionVisitor,
-      svgVisitor,
-      attrVisitor
-    )
+    visitor: {
+      Program(path: any) {
+        if (!/\.svg$/.test(state.file.opts.filename)) {
```
Comment (Andarist) [inline packages/babel-plugin-react-svg/src/index.js]: we could probably make this configurable or use better heuristic, if you have any suggestions how you'd like to have it adjusted, let me know

File: packages/babel-plugin-react-svg/src/index.js [hunk 2]
```diff
@@ -146,25 +146,30 @@ export default function(babel: BabelCore) {
...
-    visitor: Object.assign(
-      {},
-      programVisitor,
-      svgExpressionVisitor,
-      svgVisitor,
-      attrVisitor
-    )
+    visitor: {
+      Program(path: any, state: any) {
+        if (!/\.svg$/.test(state.file.opts.filename)) {
```
Comment (Andarist) [inline packages/babel-plugin-react-svg/src/index.js]: we could probably make this configurable or use better heuristic, if you have any suggestions how you'd like to have it adjusted, let me know

**Comment by boopathi** [comment]
Do you have a use case that you want it in the babel-plugin ? 

or possible to use it with tools around the plugin - cli, webpack-loader where the plugin doesn't worry about filename ?

**Comment by Andarist** [comment]
I need to use bare babel plugin, not as a webpack loader. At the moment as the plugin parse each file it crashes on some other, regular, components.

I'd like to include it in my `.babelrc` without setting up any filtering that it should only be used on my `.svg` files - imho that should be handled in the plugin itself.

**Comment by Andarist** [comment]
> or possible to use it with tools around the plugin - cli, webpack-loader where the plugin doesn't worry about filename ?

Sure thing, If you'd be open to accepting my change in some form, I'd make sure to make it work with surrounding tooling. Please let me know what you decide.

**Comment by boopathi** [comment]
a babel plugin doesn't always receive a `filename`. So I'd prefer no silent returns incase there is no filename.

So will something like this work for you -

```js
// .babelrc.js
module.exports = {
  plugins: [
    ["babel-plugin-react-svg", { extensions: [ ".svg" ] }
  ]
};
```

And, if there are no extensions, then do NOT check for filename ?

**Comment by boopathi** [comment]
Though I'm not sure if this is option `extensions` might be useful for other extension types. Since it'll always be SVG. Any option name that seems reasonable/easy to distinguish explicitly that it checks for file extension will be good.

**Comment by boopathi** [comment]
closing it as outdated. please comment to reopen

**Pull request closed without merging**